Compiling with clang on mingw64

[//]: # (Thank you for taking the time to report an issue or propose an enhancement. Questions are also welcome.)
[//]: # (If you have more than one item to report, please open separate issues!)

[//]: # (This text will not be visible and you can freely delete it. Check the Preview tab for a cleaner look.)

### What is wrong
Cannot compile JA2 with clang 16.0.1 on mingw64 environment.

$ ninja -j1
[2/287] Building CXX object CMakeFiles/ja2.dir/src/externalized/scripting/EnumCodeGen.cc.obj
FAILED: CMakeFiles/ja2.dir/src/externalized/scripting/EnumCodeGen.cc.obj
C:\msys64\clang64\bin\c++.exe -DINSTALL_LIB_DIR=\"lib\" -DMINGW_STDTHREADS_GENERATED_STDHEADERS -DWITH_UNITTESTS -IC:/msys64/home/askon/ja2-stracciatella/src -IC:/msys64/home/askon/ja2-stracciatella/src/externalized/army -IC:/msys64/home/askon/ja2-stracciatella/src/externalized/content -IC:/msys64/home/askon/ja2-stracciatella/src/externalized/internals -IC:/msys64/home/askon/ja2-stracciatella/src/externalized/json -IC:/msys64/home/askon/ja2-stracciatella/src/externalized/mercs -IC:/msys64/home/askon/ja2-stracciatella/src/externalized/policy -IC:/msys64/home/askon/ja2-stracciatella/src/externalized/scripting -IC:/msys64/home/askon/ja2-stracciatella/src/externalized/strategic -IC:/msys64/home/askon/ja2-stracciatella/src/externalized/strings -IC:/msys64/home/askon/ja2-stracciatella/src/externalized/tileset -IC:/msys64/home/askon/ja2-stracciatella/src/externalized/tactical -IC:/msys64/home/askon/ja2-stracciatella/src/externalized/ui -IC:/msys64/home/askon/ja2-stracciatella/src/externalized -IC:/msys64/home/askon/ja2-stracciatella/src/game/Editor -IC:/msys64/home/askon/ja2-stracciatella/src/game/Laptop -IC:/msys64/home/askon/ja2-stracciatella/src/game/Strategic -IC:/msys64/home/askon/ja2-stracciatella/src/game/Tactical -IC:/msys64/home/askon/ja2-stracciatella/src/game/TacticalAI -IC:/msys64/home/askon/ja2-stracciatella/src/game/TileEngine -IC:/msys64/home/askon/ja2-stracciatella/src/game/Utils -IC:/msys64/home/askon/ja2-stracciatella/src/game -IC:/msys64/home/askon/ja2-stracciatella/src/sgp -IC:/msys64/clang64/include/SDL2 -IC:/msys64/home/askon/ja2-stracciatella-build/lib-sol2/src/single/include -IC:/msys64/home/askon/ja2-stracciatella-build/lib-lua/src/src -IC:/msys64/home/askon/ja2-stracciatella/dependencies/lib-smacker/libsmacker -IC:/msys64/home/askon/ja2-stracciatella-build/lib-miniaudio/src -IC:/msys64/home/askon/ja2-stracciatella-build/lib-magic_enum/src/include -IC:/msys64/home/askon/ja2-stracciatella-build/lib-stracciatella/include -IC:/msys64/home/askon/ja2-stracciatella-build/lib-string_theory/include -IC:/msys64/home/askon/ja2-stracciatella-build/lib-mingw-std-threads/src -IC:/msys64/home/askon/ja2-stracciatella-build/lib-mingw-std-threads/build/cmake_stdheaders_generator/cmake_stdheaders_generator -isystem C:/msys64/home/askon/ja2-stracciatella-build/lib-gtest/src/googletest/include -isystem C:/msys64/home/askon/ja2-stracciatella-build/lib-gtest/src/googletest -Wall -Wtype-limits -Wignored-qualifiers -O2 -g -DNDEBUG -std=gnu++17 -MD -MT CMakeFiles/ja2.dir/src/externalized/scripting/EnumCodeGen.cc.obj -MF CMakeFiles\ja2.dir\src\externalized\scripting\EnumCodeGen.cc.obj.d -o CMakeFiles/ja2.dir/src/externalized/scripting/EnumCodeGen.cc.obj -c C:/msys64/home/askon/ja2-stracciatella/src/externalized/scripting/EnumCodeGen.cc
In file included from C:/msys64/home/askon/ja2-stracciatella/src/externalized/scripting/EnumCodeGen.cc:1:
In file included from C:/msys64/home/askon/ja2-stracciatella/src/externalized/scripting/EnumCodeGen.h:4:
C:/msys64/home/askon/ja2-stracciatella-build/lib-magic_enum/src/include/magic_enum.hpp:371:12: error: integer value -1 is outside the valid range of values [0, 255] for this enumeration type [-Wenum-constexpr-conversion]
    return static_cast<E>(static_cast<int>(i) + O);
           ^
C:/msys64/home/askon/ja2-stracciatella-build/lib-magic_enum/src/include/magic_enum.hpp:361:30: error: integer value 4294967295 is outside the valid range of values [0, 255] for this enumeration type [-Wenum-constexpr-conversion]
  return n<E, static_cast<E>(V)>().size() != 0;
                             ^
C:/msys64/home/askon/ja2-stracciatella-build/lib-magic_enum/src/include/magic_enum.hpp:335:56: error: integer value 4294967295 is outside the valid range of values [0, 255] for this enumeration type [-Wenum-constexpr-conversion]
  constexpr auto custom_name = customize::enum_name<E>(V);
                                                       ^
C:/msys64/home/askon/ja2-stracciatella-build/lib-magic_enum/src/include/magic_enum.hpp:371:12: error: integer value 257 is outside the valid range of values [0, 255] for this enumeration type [-Wenum-constexpr-conversion]
    return static_cast<E>(static_cast<int>(i) + O);
           ^
C:/msys64/home/askon/ja2-stracciatella-build/lib-magic_enum/src/include/magic_enum.hpp:361:30: error: integer value 257 is outside the valid range of values [0, 255] for this enumeration type [-Wenum-constexpr-conversion]
  return n<E, static_cast<E>(V)>().size() != 0;
                             ^
C:/msys64/home/askon/ja2-stracciatella-build/lib-magic_enum/src/include/magic_enum.hpp:335:56: error: integer value 257 is outside the valid range of values [0, 255] for this enumeration type [-Wenum-constexpr-conversion]
  constexpr auto custom_name = customize::enum_name<E>(V);
                                                       ^
C:/msys64/home/askon/ja2-stracciatella-build/lib-magic_enum/src/include/magic_enum.hpp:371:12: error: integer value 256 is outside the valid range of values [0, 255] for this enumeration type [-Wenum-constexpr-conversion]
    return static_cast<E>(static_cast<int>(i) + O);
           ^
C:/msys64/home/askon/ja2-stracciatella-build/lib-magic_enum/src/include/magic_enum.hpp:361:30: error: integer value 256 is outside the valid range of values [0, 255] for this enumeration type [-Wenum-constexpr-conversion]
  return n<E, static_cast<E>(V)>().size() != 0;
                             ^
C:/msys64/home/askon/ja2-stracciatella-build/lib-magic_enum/src/include/magic_enum.hpp:335:56: error: integer value 256 is outside the valid range of values [0, 255] for this enumeration type [-Wenum-constexpr-conversion]
  constexpr auto custom_name = customize::enum_name<E>(V);
                                                       ^
C:/msys64/home/askon/ja2-stracciatella-build/lib-magic_enum/src/include/magic_enum.hpp:335:56: error: integer value 256 is outside the valid range of values [0, 255] for this enumeration type [-Wenum-constexpr-conversion]
C:/msys64/home/askon/ja2-stracciatella-build/lib-magic_enum/src/include/magic_enum.hpp:335:56: error: integer value 256 is outside the valid range of values [0, 255] for this enumeration type [-Wenum-constexpr-conversion]
C:/msys64/home/askon/ja2-stracciatella-build/lib-magic_enum/src/include/magic_enum.hpp:371:12: error: integer value 129 is outside the valid range of values [0, 7] for this enumeration type [-Wenum-constexpr-conversion]
    return static_cast<E>(static_cast<int>(i) + O);
           ^
C:/msys64/home/askon/ja2-stracciatella-build/lib-magic_enum/src/include/magic_enum.hpp:361:30: error: integer value 129 is outside the valid range of values [0, 7] for this enumeration type [-Wenum-constexpr-conversion]
  return n<E, static_cast<E>(V)>().size() != 0;
                             ^
C:/msys64/home/askon/ja2-stracciatella-build/lib-magic_enum/src/include/magic_enum.hpp:335:56: error: integer value 129 is outside the valid range of values [0, 7] for this enumeration type [-Wenum-constexpr-conversion]
  constexpr auto custom_name = customize::enum_name<E>(V);
                                                       ^
C:/msys64/home/askon/ja2-stracciatella-build/lib-magic_enum/src/include/magic_enum.hpp:371:12: error: integer value 8 is outside the valid range of values [0, 7] for this enumeration type [-Wenum-constexpr-conversion]
    return static_cast<E>(static_cast<int>(i) + O);
           ^
C:/msys64/home/askon/ja2-stracciatella-build/lib-magic_enum/src/include/magic_enum.hpp:361:30: error: integer value 8 is outside the valid range of values [0, 7] for this enumeration type [-Wenum-constexpr-conversion]
  return n<E, static_cast<E>(V)>().size() != 0;
                             ^
C:/msys64/home/askon/ja2-stracciatella-build/lib-magic_enum/src/include/magic_enum.hpp:335:56: error: integer value 8 is outside the valid range of values [0, 7] for this enumeration type [-Wenum-constexpr-conversion]
  constexpr auto custom_name = customize::enum_name<E>(V);
                                                       ^
C:/msys64/home/askon/ja2-stracciatella-build/lib-magic_enum/src/include/magic_enum.hpp:371:12: error: integer value 9 is outside the valid range of values [0, 7] for this enumeration type [-Wenum-constexpr-conversion]
    return static_cast<E>(static_cast<int>(i) + O);
           ^
C:/msys64/home/askon/ja2-stracciatella-build/lib-magic_enum/src/include/magic_enum.hpp:361:30: error: integer value 9 is outside the valid range of values [0, 7] for this enumeration type [-Wenum-constexpr-conversion]
  return n<E, static_cast<E>(V)>().size() != 0;
                             ^
fatal error: too many errors emitted, stopping now [-ferror-limit=]
20 errors generated.
ninja: build stopped: subcommand failed.

--- a/dependencies/lib-magic_enum/getter/CMakeLists.txt.in
+++ b/dependencies/lib-magic_enum/getter/CMakeLists.txt.in
@@ -12,8 +12,8 @@ project(getter NONE)
 
 include(ExternalProject)
 externalproject_add(get-magic_enum
-    URL "https://github.com/Neargye/magic_enum/archive/v0.7.3.zip"
-    URL_MD5 "86338841251da1bacab82684a76c4042"
+    URL "https://github.com/Neargye/magic_enum/archive/v0.8.2.zip"
+    URL_MD5 "3e684af4d7073c2d86f6ecf2f78b4120"
     CMAKE_ARGS
         "-G@CMAKE_GENERATOR@"
         SOURCE_DIR          "@SRC_DIR@"

--- a/src/externalized/scripting/EnumCodeGen.h
+++ b/src/externalized/scripting/EnumCodeGen.h
@@ -54,10 +54,10 @@ void PrintEnumFlags(std::ostream& os, const ST::string& zTypeName = ST::null)
 #ifndef MAGIC_ENUM_SUPPORTED_ALIASES
 	throw std::runtime_error("enum alias support is required")
 #endif
-	os << (zTypeName.empty() ? magic_enum::flags::enum_type_name<E>() : zTypeName.to_std_string())
+	os << (zTypeName.empty() ? magic_enum::enum_type_name<E>() : zTypeName.to_std_string())
 	   << " = {" << std::endl;
 
-	constexpr auto& entries = magic_enum::flags::enum_entries<E>();
+	constexpr auto& entries = magic_enum::enum_entries<E>();
 	for (auto& pair : entries)
 	{
 		E value = pair.first;

